+++ b/models/marts/mrt_validated.sql
@@ -1,6 +1,7 @@
 with usage_data as (
     select
         usage,
+        to_char(from_timestamp, 'YYYY-MM') as month,
         injection,
         validated
     from {{ ref('fct_electricity') }}
@@ -13,6 +14,8 @@
     )
 ),
 
+peak_usage as (
+    select month, max(usage) as peak_usage from usage_data group by month
+),
 
 final as (
     select
         sum(usage) as total_usage,
@@ -20,6 +23,14 @@
         validated
     from usage_data
     group by validated
+),
+
+rolling_avg as (
+    select
+        month,
+        avg(peak_usage) over (order by month rows between 11 preceding and current row) as rolling_avg
+    from peak_usage
 )
 
 select *
-from final
+from final
+left join rolling_avg on final.month = rolling_avg.month
